### PR TITLE
Http logging

### PR DESCRIPTION
**Pull request by shashiranjan84**

### CODE REVIEW

File: kong/plugins/httplog/log.lua [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  ok, err = sock:setkeepalive(conf.keepalive)
+  if not ok then
+    ngx.log(ngx.ERR, "failed to keepalive to "..host..":"..tostring(port)..": ", err)
+    return
+  end
+end
+
+function _M.execute(conf)
+  local ok, err = ngx.timer.at(0, log, conf, ngx.ctx.log_message)
```
Comment (thibaultCha) [inline kong/plugins/httplog/log.lua]: What is the benefit of putting this into a timer?

Comment (thefosk) [inline kong/plugins/httplog/log.lua]: @thibaultCha you don't have access to the cosocket API into the `log_by_lua` handler, that's why a timer is needed.

Comment (thibaultCha) [inline kong/plugins/httplog/log.lua]: Ok

**Comment by thibaultCha** [comment]
:+1:

**Comment by sinzone** [comment]
:tada:

**Comment by shashiranjan84** [comment]
Yay

**Comment by thefosk** [comment]
Without squashing?

**Pull request merged**